lineapy/api/api.py

```diff
@@ -236,8 +234,22 @@ def _try_write_to_db(value: object) -> str:
         # Don't see an easy way to catch all possible exceptions from the to_pickle, so just catch everything for now
         logger.error(e)
         track(ExceptionEvent(ErrorType.SAVE, "Pickling error"))
-        raise e
-    return artifact_filename
+        raise ArtifactSaveException() from e
+
+
+def _try_delete_pickle_file(pickled_path: Path) -> None:
```

Comment:
I think this function is no longer used. The delete function now wraps pickle file deletion and database deletion together.